### PR TITLE
Unify matrix for split common and regular matrix

### PR DESCRIPTION
**Pull request by drashna**
## Description

This merges the split matrix.c with the regular matrix.c, since the files are nearly identical anyways.  This will make management of the matrix code simpler, in theory, as there will only be one source of truth for matrix scanning (aside from custom matrices). 

Tested on a corne, and verified that it does work properly. 

## Types of Changes

- [x] Core
- [x] Enhancement/optimization

## Issues Fixed or Closed by This PR

* Probably an item on tzarc's to-do list

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by tzarc — APPROVED** [review]

**Comment by mtei** [comment]
To my liking, it would be better to move the definition of `matrix_post_scan ()` to the new file `quantum/split_common/matrix_split.c`.  Because I think it's more natural for code that is closely related to `transport.c` to be in the same hierarchy.

**Comment by drashna** [comment]
> matrix_post_scan

Would it make sense to throw the function into `split_util.c` instead? 
Especially since there are some other functions there already that are being called?

**Review by fauxpark — APPROVED** [review]

File: quantum/matrix.c [hunk 1]
```diff
@@ -21,26 +21,60 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #include "matrix.h"
 #include "debounce.h"
 #include "quantum.h"
+#ifdef SPLIT_KEYBOARD
+#    include "split_util.h"
+#    include "config.h"
```
Comment (mtei) [inline quantum/matrix.c:26]: Is this necessary?
```
#    include "config.h"
```

Comment (drashna) [reply quantum/matrix.c:26]: Nope, it's a copy paste issue.

**Comment by mtei** [comment]
I find the following code to be a bit easier to read.

```c
#ifdef DIRECT_PINS_RIGHT
#    define SPLIT_MUTABLE
#else
#    define SPLIT_MUTABLE const
#endif
#ifdef MATRIX_ROW_PINS_RIGHT
#    define SPLIT_MUTABLE_ROW
#else
#    define SPLIT_MUTABLE_ROW const
#endif
#ifdef MATRIX_COL_PINS_RIGHT
#    define SPLIT_MUTABLE_COL
#else
#    define SPLIT_MUTABLE_COL const
#endif

#ifdef DIRECT_PINS
static SPLIT_MUTABLE pin_t direct_pins[MATRIX_ROWS][MATRIX_COLS] = DIRECT_PINS;
#elif (DIODE_DIRECTION == ROW2COL) || (DIODE_DIRECTION == COL2ROW)
#    ifdef MATRIX_ROW_PINS
static SPLIT_MUTABLE_ROW pin_t row_pins[MATRIX_ROWS] = MATRIX_ROW_PINS;
#    endif // MATRIX_ROW_PINS
#    ifdef MATRIX_COL_PINS
static SPLIT_MUTABLE_COL pin_t col_pins[MATRIX_COLS] = MATRIX_COL_PINS;
#    endif // MATRIX_COL_PINS
#endif
```

**Comment by drashna** [comment]
It's definitely a lot more concise, and I believe that tzarc mentioned doing something similar. 

Applied.

**Comment by drashna** [comment]
@mtei 
Looks like this breaks a couple of the helix boards, due to name overlapping.  
Added a commit to rename the files to prevent this issue.

**Comment by mtei** [comment]
`quantum/matrix.c` must include `quantum/split_common/split_util.h` and the other `split_util.h` must be ignored.
The best way to do this is to explicitly specify `split_common/` as follows:

```diff
diff --git a/quantum/matrix.c b/quantum/matrix.c
index 7b6954e54..ed4643f81 100644
--- a/quantum/matrix.c
+++ b/quantum/matrix.c
@@ -22,8 +22,8 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #include "debounce.h"
 #include "quantum.h"
 #ifdef SPLIT_KEYBOARD
-#    include "split_util.h"
-#    include "transactions.h"
+#    include "split_common/split_util.h"
+#    include "split_common/transactions.h"
 
 #    ifndef ERROR_DISCONNECT_COUNT
 #        define ERROR_DISCONNECT_COUNT 5
```

So you don't need to modify the files under `keyboards/helix/`.

See the URL below for the behavior of the C preprocessor `include`.
* https://gcc.gnu.org/onlinedocs/cpp/Include-Syntax.html
* https://docs.oracle.com/cd/E37069_01/html/E37074/bjadq.html#scrolltoc

**Comment by drashna** [comment]
thanks for the info and correction, @mtei!

I've reverted the commit and replaced it with this.

**Review by mtei — APPROVED** [review]

**Comment by drashna** [comment]
And now I'll see about pulling out more of the split code from the matrix, as well.

**Pull request merged**